tests/wsdiscovery.spec.ts

```diff
@@ -98,152 +175,173 @@ describe('WsDiscovery', () => {
             expect(errorMessage).to.equal(errorMsg);
         });
 
-        it('should set multicast interface to provided interface addr', () => {
-            sinon.spy(dummySocket, 'setMulticastInterface');
-            wsdd = new WsDiscovery({
-                targetInterfaceAddr: '127.0.0.1',
-            });
-
-            expect(dummySocket.setMulticastInterface).to.have.been.calledWith('127.0.0.1');
+        it('should start watching network interfaces', () => {
+            wsdd = new WsDiscovery(wsddOptions);
+            expect(wsdd.networkInterfacesWatcher).to.not.equal(undefined);
         });
+    });
 
-        it('should use base interface address by default', () => {
-            sinon.spy(dummySocket, 'setMulticastInterface');
-
-            wsdd = new WsDiscovery();
-
-            expect(dummySocket.setMulticastInterface).to.have.been.calledWith('169.254.158.54');
+    describe('#bindSocket', () => {
+        let socket, bindSpy;
+        beforeEach(() => {
+            wsdd = new WsDiscovery(wsddOptions);
+            socket = new DummySocket();
+            bindSpy = sinon.spy(socket, 'bind');
+        });
+        afterEach(() => {
+            bindSpy.restore();
+        });
+        it('should bind socket to provided ip address', () => {
+            wsdd.bindSocket({ ip: '0.0.0.0', interfaceName: 'A name' }, socket);
+            expect(bindSpy).to.have.been.calledOnce;
+            expect(bindSpy).to.have.been.calledWith({ address: '0.0.0.0' });
         });
 
-        describe('watchInterface', () => {
-            let watchIfSpy;
-            beforeEach(() => {
-                watchIfSpy = sinon.spy(WsDiscovery.prototype, 'watchInterface');
-            });
-            afterEach(() => {
-                watchIfSpy.restore();
-            });
-            it('should call watchInterface for default when no BASE device present', () => {
-                networkInterfacesStub.returns({});
-                wsdd = new WsDiscovery();
-                expect(watchIfSpy.called).to.equal(true);
-                expect(watchIfSpy.getCall(0).args[0]).to.equal('default');
-            });
-            it('should call watchInterface for BASE interface', () => {
-                wsdd = new WsDiscovery();
-                expect(watchIfSpy.called).to.equal(true);
-                expect(watchIfSpy.getCall(0).args[0]).to.equal('baseKey');
-            });
+        it('should bind socket to the whole network when no ip address is provided', () => {
```

Comment:
Good that u are testing both when ip is provided and when it is undefined. To make sure the bind is called only once.